api/v1beta1/ocicluster_webhook.go

```diff
@@ -24,19 +24,27 @@ import (
 
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/validation/field"
-
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
 var clusterlogger = ctrl.Log.WithName("ocicluster-resource")
 
 var (
+	_ webhook.Defaulter = &OCICluster{}
 	_ webhook.Validator = &OCICluster{}
 )
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-ocicluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=ociclusters,versions=v1beta1,name=validation.ocicluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ocicluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=ociclusters,versions=v1beta1,name=default.ocicluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
+
+func (c *OCICluster) Default() {
+	if c.Spec.OCIResourceIdentifier == "" {
+		c.Spec.OCIResourceIdentifier = string(uuid.NewUUID())
+	}
+}
```

Comment:
Do we need a test to make sure this is set?

Reply:
Also should we some how validate this exists during create? I believe the mutation happens followed by verification.